packages/react/src/components/AccountSettings/ChangePassword/ChangePassword.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+
+import { Logger } from 'aws-amplify';
+import { changePassword, translate } from '@aws-amplify/ui';
+
+import { useAuth } from '../../../internal';
+import { View, Flex } from '../../../primitives';
+import {
+  DefaultCurrentPassword,
+  DefaultError,
+  DefaultNewPassword,
+  DefaultSubmitButton,
+} from './defaultComponents';
+import { ChangePasswordProps } from './types';
+import { FormValues } from '../types';
+
+const logger = new Logger('ChangePassword');
+
+const ChangePassword: React.ComponentType<ChangePasswordProps> = (props) => {
+  const { onSuccess, onError } = props;
+  const [error, setError] = React.useState<string>(null);
+  const [formValues, setFormValues] = React.useState<FormValues>({});
+
+  const { user, isLoading: isAuthUserLoading } = useAuth();
+
+  /** Translations */
+  // TODO: add AccountSettingsTextUtil to collect these strings
+  const currentPasswordLabel = translate('Current Password');
+  const newPasswordLabel = translate('New Password');
+  const updatePasswordText = translate('Update password');
+
+  /** Auth API Handler */
+  const handleChangePassword = React.useCallback(async () => {
+    const { currentPassword, newPassword } = formValues;
+    try {
+      await changePassword({ user, currentPassword, newPassword });
+      setError(null);
+
+      if (onSuccess) {
+        onSuccess(); // notify success to the parent
+      }
+    } catch (e) {
+      const error = e as Error;
+      setError(error.message);
+
+      if (onError) {
+        onError(error); // notify error to the parent
+      }
+    }
+  }, [formValues, user, onSuccess, onError]);
+
+  /** Event Handlers */
+  const handleChange = React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      event.preventDefault();
+
+      const { name, value } = event.target;
+      setFormValues({ ...formValues, [name]: value });
+    },
+    [formValues]
+  );
+
+  const handleSubmit = React.useCallback(
+    (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      handleChangePassword();
+    },
+    [handleChangePassword]
+  );
+
+  // return null if Auth.getCurrentAuthenticatedUser is still in progress
+  if (isAuthUserLoading) {
+    return null;
+  }
+
+  if (!user) {
+    logger.warn('<ChangePassword /> requires user to be authenticated.');
+    return null;
+  }
+
+  return (
+    <View as="form" className="amplify-changepassword" onSubmit={handleSubmit}>
+      <Flex direction="column">
```

Comment:
We could simplify these 2 lines I think:
```suggestion
    <Flex as="form" direction="column" className="amplify-changepassword" onSubmit={handleSubmit}>
```

Reply:
I've initially tried this, but `onSubmit` handler for `Flex as="form"` had TS errors:

```
Type '(event: React.FormEvent<HTMLFormElement>) => void' is not assignable to type 'FormEventHandler<HTMLDivElement>'.
```

I'll create a repro and submit an issue. I think we can address this later after then.

Reply:
FYI created #2891